WordPress/gutenberg
# Fix toolbar keyboard navigation after pressing alt+F10

Closes #28016

The specific block's toolbar items (`Change text alignment`, `Bold` etc. for Paragraph) are added through the `SlotFill` module, which uses React Portal. The block popover has a `Slot` component, whose node serves as the container for the portal. React events only propagate when elements are in the same React subtree, which doesn't happen in this case.

When those elements receive focus, React's `onFocus` and `onBlur` will not propagate, so the popover thinks they're outside. In fact, those elements are outside of the popover in the React tree, so this makes sense.

~Had to convert those events to DOM ones. But I'm open to suggestions. I thought about changing it directly on the `useFocusOutside` hook, but it would be more complicated since this is a public API.~

This PR removes the `onFocusOutside` behavior and uses a logic similar to when we press <kbd>Shift+Tab</kbd>.

## Screenshots

<img src="https://user-images.githubusercontent.com/3068563/104082833-1fbb6680-5218-11eb-878d-8672277b3542.gif" width="400" alt="Navigating through the block toolbar after pressing alt+F10" />

## How has this been tested?

- Check the added e2e test.
- Use the block toolbar and check if #28016 is fixed and the toolbar behaves correctly.

## Types of changes

Bug fix

## Checklist:
